synapse/handlers/message.py

```diff
@@ -644,7 +644,7 @@ async def send_nonmember_event(
         event: EventBase,
         context: EventContext,
         ratelimit: bool = True,
-    ) -> int:
+    ) -> Union[int, EventBase]:
```

Comment:
This feels like a bug that we should fix rather than document?

Reply:
I agree. I wasn't sure it made sense to tie up in this PR though. I'll probably split out. Do you have any opinion about what this should return in the "error" case?

Reply:
See #8093.